include/boost/gil/extension/toolbox/image_types/subchroma_image.hpp

```diff
@@ -23,6 +26,46 @@
 
 namespace boost{ namespace gil {
 
+/////////////////////////////////////////////////////////////
+template< int J
+        , int a
+        , int b
+        >
+struct Scaling_Factors
+{
+    BOOST_STATIC_ASSERT(( mpl::equal_to< mpl::int_< J >, mpl::int_< 4 > >::type::value ));
+
+    BOOST_STATIC_ASSERT(( mpl::or_< mpl::equal_to< mpl::int_< a >, mpl::int_< 4 > >
+                                  , mpl::or_< mpl::equal_to< mpl::int_< a >, mpl::int_< 2 > >
+                                            , mpl::equal_to< mpl::int_< a >, mpl::int_< 1 > >
+                                            >
+                                  >::type::value
+                       ));
+
+    BOOST_STATIC_ASSERT(( mpl::or_< mpl::equal_to< mpl::int_< b >, mpl::int_< 4 > >
+                                  , mpl::or_< mpl::equal_to< mpl::int_< b >, mpl::int_< 2 > >
+                                            , mpl::or_< mpl::equal_to< mpl::int_< b >, mpl::int_< 1 > >
+                                                      , mpl::equal_to< mpl::int_< b >, mpl::int_< 0 > >
+                                                      >
+                                            >
+                                  >::type::value
+                       ));
+
+    BOOST_STATIC_CONSTANT( int, ss_X = ( mpl::divides< mpl::int_< J >
+                                                     , mpl::int_< a >
+                                                     >::type::value )
+                         );
+
+    BOOST_STATIC_CONSTANT( int, ss_Y = ( mpl::if_< mpl::equal_to< mpl::int_< b >, mpl::int_< 0 > >
+                                                 , mpl::int_< 2 >
+                                                 , mpl::if_< mpl::equal_to< mpl::int_< a >, mpl::int_< b > >
+                                                           , mpl::int_< 1 >
+                                                           , mpl::int_< 4 >
+                                                           >
+                                                 >::type::value )
+                         );
+};
+
```

Comment:
It looks like this is only used as a helper type inside the class below. I would therefore suggest to move this into a `detail` namespace to "hide" it from the public API.

Reply:
Good idea. 

I have also renamed `Scaling_Factors` to lower-case, as well as its template parameters.

The header test passes now:

```
mloskot@bionic:/mnt/d/boost.wsl/libs/gil$ colorbb -j 8 -q test/headers//extension/toolbox
Performing configuration checks

    - default address-model    : 64-bit (cached)
    - default architecture     : x86 (cached)
    - symlinks supported       : yes (cached)
...patience...
...found 1801 targets...
...updating 153 targets...
...
gcc.compile.c++ ../../bin.v2/libs/gil/test/headers/h-extension-toolbox-image_types-subchroma_image.test/gcc-5.5.0/debug/visibility-hidden/h-extension-toolbox-image_types-subchroma_image.o
**passed** ../../bin.v2/libs/gil/test/headers/h-extension-toolbox-image_types-subchroma_image.test/gcc-5.5.0/debug/visibility-hidden/h-extension-toolbox-image_types-subchroma_image.test
```